pkgs/build-support/fetchpatch/default.nix

```
@@ -13,15 +13,20 @@ in
 
 let
   # Make base-64 encoded SRI hash filename-safe using RFC 4648 §5
-  tmpname = lib.replaceStrings [ "+" "/" "=" ] [ "-" "_" "" ] args.sha256;
+  tmpname = lib.replaceStrings [ "+" "/" "=" ] [ "-" "_" "" ] args.sha256 or "";
 in
 fetchurl ({
   postFetch = ''
     tmpfile="$TMPDIR/${tmpname}"
+
+    # Cleanly exit if sha256 isn't correctly set, to display fetchurl's error message.
+    [[ "$tmpfile" = "$NIX_BUILD_TOP/" ]] && exit 0
```

Huh... Isn't it better to do this check as:

```suggestion
    # Cleanly exit if sha256 isn't correctly set, to display fetchurl's error message.
    [[ -z "${tmpname}" ]] && exit 1

    tmpfile="$TMPDIR/${tmpname}"
```

Also, changing `exit 0` to `exit 1` to indicate that this is an error.

---

> Huh... Isn't it better to do this check as:

It is! Thanks :)

> Also, changing exit 0 to exit 1 to indicate that this is an error.

Doing this causes the entire builder to quit, so `fetchurl`'s error message isn't shown.